src/main/java/mSearch/filmlisten/FilmlisteLesen.java

```diff
@@ -119,125 +82,187 @@ private InputStream selectDecompressor(String source, InputStream in) throws Exc
+        } else {
+            milliseconds = 0;
+        }
+    }
 
-        if (Config.getStop()) {
-            Log.sysLog("--> Abbruch");
+    public void readFilmListe(String source, final ListeFilme listeFilme, int days) {
+        try {
+            Log.sysLog("Liste Filme lesen von: " + source);
             listeFilme.clear();
+            this.notifyStart(source, PROGRESS_MAX); // für die Progressanzeige
+
+            checkDays(days);
+
+            if (!source.startsWith("http")) {
+                processFromFile(source, listeFilme);
+            } else {
+                processFromWeb(new URL(source), listeFilme);
+            }
+
+            if (Config.getStop()) {
+                Log.sysLog("--> Abbruch");
+                listeFilme.clear();
+            }
+        } catch (MalformedURLException ex) {
+            ex.printStackTrace();
         }
+
         notifyFertig(source, listeFilme);
     }
 
+    /**
+     * Download a process a filmliste from the web.
+     *
+     * @param source     source url as string
+     * @param listeFilme the list to read to
+     */
+    private void processFromWeb(URL source, ListeFilme listeFilme) {
+        Request.Builder builder = new Request.Builder().url(source);
+        builder.addHeader("User-Agent", Config.getUserAgent());
+
+        //our progress monitor callback
+        InputStreamProgressMonitor monitor = new InputStreamProgressMonitor() {
+            private int oldProgress = 0;
+
+            @Override
+            public void progress(long bytesRead, long size) {
+                final int iProgress = (int) (bytesRead * 100 / size);
+                if (iProgress != oldProgress) {
+                    oldProgress = iProgress;
+                    notifyProgress(source.toString(), iProgress);
+                }
+            }
+        };
+
+        try (Response response = MVHttpClient.getInstance().getHttpClient().newCall(builder.build()).execute();
+             ResponseBody body = response.body()) {
+            if (response.isSuccessful()) {
+                try (InputStream input = new ProgressMonitorInputStream(body.byteStream(), body.contentLength(), monitor)) {
+                    try (InputStream is = selectDecompressor(source.toString(), input);
+                         JsonParser jp = new JsonFactory().createParser(is)) {
+                        readData(jp, listeFilme);
+                    }
+                }
+            }
+        } catch (Exception ex) {
+            Log.errorLog(945123641, ex, "FilmListe: " + source);
+            listeFilme.clear();
+        }
+    }
+
     private boolean checkDate(DatenFilm film) {
         // true wenn der Film angezeigt werden kann!
         try {
             if (film.datumFilm.getTime() != 0) {
-                if (film.datumFilm.getTime() < seconds) {
+                if (film.datumFilm.getTime() < milliseconds) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1066)